src/picongpu/include/fields/MaxwellSolver/Lehe/LeheCurl.hpp

```diff
@@ -176,59 +185,66 @@ namespace picongpu
             typedef PMacc::math::CT::Int< 1, 1, 1 > LowerMargin;
             typedef PMacc::math::CT::Int< 2, 2, 2 > UpperMargin;
 
+            float_X mySin;
+
+            HINLINE CurlELehe( )
+            {
+                mySin = float_X(
+                    math::sin(
+                        float_64( 0.5 ) *
+                        float_64( M_PI ) *  float_64( SPEED_OF_LIGHT ) *
+                        float_64( DELTA_T ) / float_64( CELL_HEIGHT )
+                    )
+                );
+            }
+
             template<class Memory >
             HDINLINE typename Memory::ValueType operator( )(const Memory & mem ) const
             {
                 /* Distinguished direction where the numerical Cherenkov Radiation
                  * of moving particles is suppressed.
                  */
-                const float_X isDir_x = float_X( 0.0 );
-                const float_X isDir_y = float_X( 1.0 );
-                const float_X isDir_z = float_X( 0.0 );
+                constexpr float_X isDir_x = float_X( 0.0 );
```

Comment:
ah wait, the sine is outside now and those constexpr should work. lets check the registers if they pop :)